doc/making-a-visualizer.md

```diff
@@ -289,3 +250,72 @@ In a documentation block just before the first event handling method, put
 You can organize the control documentation comments in whatever way makes the
 most sense. If possible, order the event handling methods so that most
 controls can be documented near the method that implements them.
+
+### Abstract visualizers
+
+Now that we've seen how to extend the [`P5Visualizer`](#p5-visualizers) base
+class, let's take a peek at how the base class works internally. This section
+will be most useful to you if you want to write a new base class, or to build
+a visualizer so different from anything else that it shouldn't have a base
+class.
```

Comment:
Suggest adding to this paragraph something along the lines "Note that you can override these methods in a P5Visualizer, too, once you understand what they do, if you want to alter how a P5Visualizer displays or operates in a more fundamental way than the usual `setup()`, `draw()`, and event-handling methods allow."

Reply:
Done.